src/onepiece_classify/infer/recognition.py

```diff
@@ -39,10 +54,19 @@ def __init__(self, model_path: str):
 
     def _build_model(self):
         # load model
-        state_dict = torch.load(self.model_path)
+        state_dict = torch.load(self.model_path, map_location=self.device)
         model_backbone = image_recog(self.nclass)
         model_backbone.load_state_dict(state_dict)
         return model_backbone
+    
+    def _get_cache_dir(self):
+        if sys.platform.startswith("win"):
```

Comment:
remember to give information in your readme for support auto cache only in linux and windows